### PR TITLE
Various monorepo project tasks

### PR DESCRIPTION
**Pull request by Invertisment**
Contains dummy TS sources for testing purposes.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# EditorConfig is awesome: https://EditorConfig.org
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+indent_style = space
+indent_size = 2
+insert_final_newline = true
+trim_trailing_whitespace = false
```
Comment (robert-zaremba) [inline .editorconfig:10]: - [ ] Please add here:

```
max_line_length = 96
```

File: package.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "version": "0.0.1",
   "description": "Framework to automate development of Algorand Assets and Smart Contracts.",
   "repository": "https://github.com/scale-it/algorand-builder.git",
-  "author": "Robert Zaremba <https://github.com/robert-zaremba>",
+  "author": "Robert Zaremba <https://github.com/robert-zaremba>, Martynas Maciulevičius <https://github.com/Invertisment>",
```
Comment (robert-zaremba) [inline package.json:6]: I think we need to move this things to `contirbutors` https://flaviocopes.com/package-json/#contributors

File: package.json [hunk 3]
```diff
@@ -15,18 +15,27 @@
...
     "chai": "^4.2.0",
     "eslint-plugin-simple-import-sort": "^5.0.3",
     "mocha": "^8.0.1",
+    "nyc": "^14.1.0",
+    "prettier": "2.0.5",
+    "ts-node": "^8.1.0",
+    "tslint": "^5.16.0",
+    "tslint-config-prettier": "^1.18.0",
+    "tslint-consistent-codestyle": "^1.15.1",
+    "tslint-plugin-prettier": "^2.0.1",
```
Comment (robert-zaremba) [inline package.json:38]: we don't need `prettier`

Also, later on we will need to remove `tslint`

File: packages/builder-core/.mocharc.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
+  "diff": true,
+  "extension": ["js", "ts"],
+  "package": "./package.json",
+  "reporter": "spec",
+  "slow": 75,
+  "timeout": 2000,
+  "ui": "bdd",
+  "watch-files": ["lib/**/*.js", "test/**/*.js", "lib/**/*.ts", "test/**/*.ts"],
```
Comment (robert-zaremba) [inline packages/builder-core/.mocharc.json:9]: shouldn't we have here `src` instead of `lib`?

File: packages/builder-core/package.json [hunk 5]
```diff
@@ -0,0 +1,136 @@
+{
+  "name": "builder-core",
+  "version": "0.0.1",
+  "author": "Robert Zaremba <https://github.com/robert-zaremba>, Martynas Maciulevičius <https://github.com/Invertisment>",
+  "license": "SEE LICENSE IN LICENSE",
```
Comment (robert-zaremba) [inline packages/builder-core/package.json:5]: it should say that license is in root

File: packages/builder-core/package.json [hunk 6]
```diff
@@ -0,0 +1,136 @@
+{
+  "name": "builder-core",
+  "version": "0.0.1",
+  "author": "Robert Zaremba <https://github.com/robert-zaremba>, Martynas Maciulevičius <https://github.com/Invertisment>",
```
Comment (robert-zaremba) [inline packages/builder-core/package.json:4]: let's remove

File: packages/builder-core/package.json [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+  "license": "SEE LICENSE IN LICENSE",
+  "homepage": "",
+  "repository": "github:scale-it/algorand-builder",
+  "main": "internal/lib/buidler-lib.js",
```
Comment (robert-zaremba) [inline packages/builder-core/package.json:8]: why `internal`?

File: packages/builder-core/package.json [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    "smart-contracts",
+    "buidler",
+    "blockchain",
+    "dapps",
+    "javascript",
+    "tooling",
+    "task-runner"
+  ],
+  "bin": {
+    "buidler": "internal/cli/cli.js"
```
Comment (robert-zaremba) [inline packages/builder-core/package.json:21]: this should be called `algob`

File: packages/builder-core/package.json (hunk 7 above)
Comment (robert-zaremba) [reply packages/builder-core/package.json:8]: Also filename is wrong. should be `algob.js` (or .ts)

File: packages/builder-core/package.json [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+    "raw-body": "^2.4.1",
+    "semver": "^6.3.0",
+    "slash": "^3.0.0",
+    "solc": "0.6.8",
+    "source-map-support": "^0.5.13",
+    "ts-essentials": "^2.0.7",
+    "tsort": "0.0.1",
+    "uuid": "^3.3.2",
+    "ws": "^7.2.1"
+  },
```
Comment (robert-zaremba) [inline packages/builder-core/package.json:112]: we need to remove all ethereum specific things (ethereum, solc, solidity, ws...)

File: packages/builder-core/src/tsconfig.json [hunk 10]
```diff
@@ -0,0 +1,9 @@
+{
+  "compilerOptions": {
+    "outDir": "../build/",
+    "rootDirs": ["."],
+    "composite": true
+  },
+  "include": ["./**/*.ts"],
+  "exclude": []
+}
```
Comment (robert-zaremba) [inline packages/builder-core/src/tsconfig.json:9]: let's move this file one level up

File: packages/builder-core/tsconfig.json [hunk 11]
```diff
@@ -0,0 +1,14 @@
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "outDir": "./build-test",
+    "rootDirs": ["./test"]
+  },
+  "include": ["./test/**/*.ts"],
+  "exclude": ["./test/**/buidler.config.ts"],
```
Comment (robert-zaremba) [inline packages/builder-core/tsconfig.json:8]: wrong name

File: tsconfig.json [hunk 12]
```diff
@@ -8,7 +8,7 @@
     "allowJs": true,
     "skipLibCheck": true,
     "esModuleInterop": true,
-    "module": "esnext",
+    "module": "commonjs",
```
Comment (robert-zaremba) [inline tsconfig.json:11]: why changing it?

File: packages/builder-core/package.json [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+    "src/",
+    "*.d.ts",
+    "*.d.ts.map",
+    "*.js",
+    "*.js.map",
+    "LICENSE",
+    "README.md",
+    "recommended-gitignore.txt",
+    "console.sol"
+  ],
```
Comment (robert-zaremba) [inline packages/builder-core/package.json:50]: - [ ] let's remove the whole `files` section.

File: packages/builder-core/src/tsconfig.json (hunk 10 above)
Comment (Invertisment) [reply packages/builder-core/src/tsconfig.json:9]: I found that linting setup uses these both configs separately. It's possible to run the linter for src and for tests separately. I'm not sure if it's useful right away.

File: tsconfig.json (hunk 12 above)
Comment (Invertisment) [reply tsconfig.json:11]: This causes these kinds of errors:
```
import { assert } from "chai";
^^^^^^

SyntaxError: Cannot use import statement outside a module```

File: packages/builder-core/package.json [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+  "dependencies": {
+    "ansi-escapes": "^4.3.0",
+    "chalk": "^2.4.2",
+    "cross-env": "5.0.5",
+    "fs-extra": "^7.0.1",
+    "glob": "^7.1.3",
+    "raw-body": "^2.4.1",
+    "semver": "^6.3.0",
+    "source-map-support": "^0.5.13",
+    "ts-essentials": "^2.0.7"
```
Comment (robert-zaremba) [inline packages/builder-core/package.json:52]: I think 3 above dependencies should be in `devDependencies`

File: packages/builder-core/tsconfig.json [hunk 15]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "outDir": "./build",
+    "rootDirs": ["./src"]
```
Comment (robert-zaremba) [inline packages/builder-core/tsconfig.json:5]: why do we need to specify `rootDirs`? I think `baseUrl` is preferred.

File: tsconfig.json [hunk 16]
```diff
@@ -8,15 +8,15 @@
...
     "resolveJsonModule": true,
-    "isolatedModules": true,
     "allowSyntheticDefaultImports": true,
     "noImplicitReturns": true,
     "strict": true,
     "forceConsistentCasingInFileNames": true,
     "sourceMap": true,
-    "baseUrl": "src"
+    "baseUrl": "src",
```
Comment (robert-zaremba) [inline tsconfig.json:18]: maybe removing `baseUrl` from here and moving it to subproject will do the 'job'

File: packages/builder-core/package.json (hunk 14 above)
Comment (Invertisment) [reply packages/builder-core/package.json:52]:
```
  "devDependencies": {
    "@types/chai": "^4.2.0",
    "chai": "^4.2.0",
    "time-require": "^0.1.2",
    "cross-env": "5.0.5",
    "source-map-support": "^0.5.13"
  },
  "dependencies": {
  },
```

File: packages/builder-core/tsconfig.json (hunk 15 above)
Comment (Invertisment) [reply packages/builder-core/tsconfig.json:5]: changing to baseUrl: src

File: package.json (hunk 3 above)
Comment (Invertisment) [reply package.json]: Removed

File: packages/builder-core/package.json (hunk 5 above)
Comment (Invertisment) [reply packages/builder-core/package.json]: apache

File: packages/builder-core/tsconfig.json (hunk 15 above)
Comment (Invertisment) [reply packages/builder-core/tsconfig.json:5]: Removed. Root project config is extended.

File: packages/builder-core/src/tsconfig.json (hunk 10 above)
Comment (Invertisment) [reply packages/builder-core/src/tsconfig.json]: builder-core/tsconfig.json compilation file was changed to compile everything instead of tests

**Pull request merged**